scopesim/effects/data_container.py

```diff
@@ -189,4 +197,121 @@ def validate(self, etype):
         etype_colname = utils.real_colname("ETYPE", self.meta.colnames)
         return self.meta[etype_colname] == etype
 
+    @property
+    def plot_filename(self):
+        plot_fname = self.meta["report"]["plot_filename"]
+        if plot_fname is None:
+            plot_fname = self.meta["name"].lower().replace(" ", "_")
+        plot_fformat = self.meta["report"]["plot_file_format"]
+
+        return plot_fname + "." + plot_fformat
+
+
+    @property
+    def table_string(self):
+        if isinstance(self.table, Table):
+            tbl_str = str(self.table).replace("-", "=")
+            hdr = tbl_str.split("\n")[1]
+            tbl_str = hdr + "\n" + tbl_str + "\n" + hdr
+        else:
+            tbl_str = ""
+
+        return tbl_str
+
+    @property
+    def meta_string(self):
+        meta_str = ""
+        max_key_len = max([len(key) for key in self.meta.keys()])
+        for key in self.meta:
+            if key not in ["comments", "changes", "description", "history", "report"]:
+                meta_str += f"    {key.rjust(max_key_len)} : {self.meta[key]}\n"
+
+        return meta_str
+
+    @property
+    def class_description(self):
+        cls_str = ""
+        if hasattr(str, "__doc__"):
+            cls_str = self.__doc__.split("\n")[0]
+
+        return cls_str
+
+    @property
+    def changes_str(self):
+        changes_str = ""
+        if "comments" in self.meta:
+            if "changes" in self.meta["comments"]:
+                for line in self.meta["comments"]["changes"]:
+                    changes_str += line
+
+        return changes_str
```

Comment:
Nitpicking: Is this the same?
```suggestion
        return "".join(self.meta.get("comments", {}).get("changes", []))
```

Reply:
How did I never come accross the get() method before now!?!?! =D =D